cli/wallet.go

```diff
@@ -235,3 +237,105 @@ var walletImport = &cli.Command{
 		return nil
 	},
 }
+
+var walletSign = &cli.Command{
+	Name:      "sign",
+	Usage:     "sign a message",
+	ArgsUsage: "<hexMessage>",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "from",
+			Usage: "signing address",
+		},
+	},
+	Action: func(cctx *cli.Context) error {
+		api, closer, err := GetFullNodeAPI(cctx)
+		if err != nil {
+			return err
+		}
+		defer closer()
+		ctx := ReqContext(cctx)
+
+		if !cctx.Args().Present() {
+			return fmt.Errorf("must specify message to sign")
+		}
+
+		var addr address.Address
+		if cctx.IsSet("from") {
+			addr, err = address.NewFromString(cctx.String("from"))
+		} else {
+			addr, err = api.WalletDefaultAddress(ctx)
+		}
+
+		if err != nil {
+			return err
+		}
+
+		msg, err := hex.DecodeString(cctx.Args().First())
+
+		if err != nil {
+			return err
+		}
+
+		sig, err := api.WalletSign(ctx, addr, msg)
+
+		if err != nil {
+			return err
+		}
+
+		sigBytes := append([]byte{byte(sig.TypeCode())}, sig.Data...)
+
+		fmt.Println(hex.EncodeToString(sigBytes))
+		return nil
+	},
+}
+
+var walletVerify = &cli.Command{
+	Name:      "verify",
+	Usage:     "verify the signature of a message",
+	ArgsUsage: "<signing address> <hexMessage> <signature>",
+	Action: func(cctx *cli.Context) error {
+		api, closer, err := GetFullNodeAPI(cctx)
+		if err != nil {
+			return err
+		}
+		defer closer()
+		ctx := ReqContext(cctx)
+
+		if !cctx.Args().Present() || cctx.NArg() != 3 {
+			return fmt.Errorf("must specify signing address, mnessage, and signature to verify")
+		}
+
+		addr, err := address.NewFromString(cctx.Args().First())
+
+		if err != nil {
+			return err
+		}
+
+		msg, err := hex.DecodeString(cctx.Args().Get(1))
+
+		if err != nil {
+			return err
+		}
+
+		sigBytes, err := hex.DecodeString(cctx.Args().Get(2))
+
+		if err != nil {
+			return err
+		}
+
+		sig, err := types.SignatureFromBytes(sigBytes)
+
+		if err != nil {
+			return err
+		}
+
+		if api.WalletVerify(ctx, addr, msg, &sig) {
+			fmt.Println("Valid signature")
+		} else {
+			fmt.Println("Invalid signature")
```

Comment:
I'd make this just `valid/invalid` (all lowercase) as matching spaces in shell scripts is a bit awkward

It would also be nice if the command exited with non-0 code (define global error like `ErrCommandFalied` somewhere in `cli/`, then in `cmd/lotus/amin.go` look for it, and `os.exit(1)` if we see it there - we don't want to just os.exit here, as that wouldn't close the RPC connection cleanly)